e2etest/zt_resume_test.go

```diff
@@ -125,3 +128,52 @@ func TestResume_PublicSource_ContainerTarget(t *testing.T) {
 		EAccountType.Standard(), EAccountType.Standard(), "",
 	)
 }
+
+// TestResume_ConsistentLogs_PublicSource_BlobTarget validates log file consistency for a job initially initiated as a copy operation and later resumed.
+// During the copy operation, a jobId is created and associated with its log file. If a resume operation follows, the same jobId from the copy is used for the log file, and the resumed logs are appended to it.
+// The test ensures that at most two log files are generated: one for scanning logs and one for normal logs.
+
+func TestResume_ConsistentLogs_PublicSource_BlobTarget(t *testing.T) {
+	RunScenarios(
+		t,
+		eOperation.Copy()|eOperation.Resume(),
+		eTestFromTo.Other(common.EFromTo.BlobBlob(), common.EFromTo.BlobLocal()),
+		eValidate.Auto(),
+		anonymousAuthOnly,
+		anonymousAuthOnly,
+		params{
+			recursive:      true,
+			debugSkipFiles: []string{";"}, // skip the only file is ;
+		},
+		&hooks{
+			afterValidation: func(h hookHelper) {
+				a := h.GetAsserter()
+				logDir := os.Getenv("AZCOPY_E2E_LOG_OUTPUT")
+				dir, err := os.Open(logDir)
+				a.Assert(err, equals(), nil)
+				defer dir.Close()
+
+				entries, err := dir.Readdir(-1)
+				a.Assert(err, equals(), nil)
+				var logFiles int
+				for _, entry := range entries {
+					if !entry.IsDir() && strings.HasSuffix(entry.Name(), ".log") {
+						logFiles++
+					}
+				}
+				a.Assert(logFiles <= 2, equals(), true)
```

Comment:
If its resume job, shouldn't there be only one log file. with <= 2 we are accepting 2 log files as well?

Reply:
In totality - there would be a scanning log-file and a normal log file pre-pended by the JobID hence we are checking for two log files